Mage/src/main/java/mage/abilities/effects/common/continuous/GainAbilityTargetEffect.java

```diff
@@ -52,8 +53,7 @@ public GainAbilityTargetEffect(Ability ability, Duration duration, String rule,
 
     protected GainAbilityTargetEffect(final GainAbilityTargetEffect effect) {
         super(effect);
-        this.ability = effect.ability.copy();
-        this.ability.newId(); // This is needed if the effect is copied e.g. by a clone so the ability can be added multiple times to permanents
```

Comment:
Please don't remove the comment, in fact; it needs to be expanded with explanation